### PR TITLE
Checks that the backup files have been created successfully by looking for specific sentinel files (VERSION.txt)

### PR DESCRIPTION
**Pull request by danielmaclaren**
As this Client_E4 folder is rather large, this process can take a while to run. An alternative method is listed below to allow users to test the function in a much faster time frame: 

1. Go to C:\Instrument\Apps
2. Rename "EPICS", "Client_E4" and "Python3" to "EPICS.backup", "Client_E4.backup" and "Python3.backup"
3. Create three new folders named "EPICS", "Client_E4" and "Python3"
4. Create a VERSION.txt file in each folder. During the test, you should remove and add other files in each folder to test if the function will still run as intended.
5. Run the 'instrument_deploy.bat' file in a command window (which can be found in "C:\Instrument\Dev\ibex_utils\installation_and_upgrade")
6. Answer 'yes' to the following prompts:

'Would you like to use Server version: 12.0.1, Client version: 12.0.2, and Genie Python version: 12.0.1?'

 Upgrade NDW2555 as a Client/Server Machine
    Server source: \\isis.cclrc.ac.uk\inst$\Kits$\CompGroup\ICP\Releases\12.0.1\EPICS
    Client source: \\isis.cclrc.ac.uk\inst$\Kits$\CompGroup\ICP\Releases\12.0.2\Client
    Python 3 source: \\isis.cclrc.ac.uk\inst$\Kits$\CompGroup\ICP\Releases\12.0.1\genie_python_3
Continue? [Y/N]

Answer 'no' to the proceeding 'Do step' lines until 'Do step 'Backup old directories'? :'

There may be an optional prompt amongst the 'Do step' lines: 
Would you like to use Server version: 12.0.1, Client version: 12.0.2, and Genie Python version: 12.0.1?
Type Y when done.

7. Press 'yes' for 'Do step 'Verify backup'? :', this should check if the backup files contain the VERSION.txt file and prompt the user if there is no file detected, which should indicate an issue with the backup. Users are allowed to proceed if they want, but a warning will be displayed and location of the missing file.

### CODE REVIEW

**Comment by danielmaclaren** [comment]
## Link to Ticket

https://github.com/ISISComputingGroup/IBEX/issues/7611

File: installation_and_upgrade/ibex_install_utils/tasks/backup_tasks.py [hunk 1]
```diff
@@ -64,6 +65,27 @@ def backup_old_directories(self):
...
+
+        """
+        EPICS_PATH_BACKUP = os.path.join(self._get_backup_dir(),'EPICS')
+        PYTHON3_PATH_BACKUP = os.path.join(self._get_backup_dir(),'Python3')
+        GUI_PATH_BACKUP = os.path.join(self._get_backup_dir(),'Client_E4')  
+
+        backup_paths = (EPICS_PATH_BACKUP, PYTHON3_PATH_BACKUP, GUI_PATH_BACKUP)
+
+        for d in backup_paths:
+            version_list = glob.glob(os.path.join(d, "VERSION.txt"))
```
Comment (FreddieAkeroyd) [inline installation_and_upgrade/ibex_install_utils/tasks/backup_tasks.py:82]: glob is normally used with a wildcard to match several files, maybe  `os.path.is_file()` is better here?

**Review by NikolaRoev — CHANGES_REQUESTED** [review]
We probably want to make the verification check in the existing backup task, and then we can just warn the user that it failed just like `_backup_dir` does it.

Checking the deploy log on LET we can see that there was no output for the client backup at all, consider adding an `else` statement to `_backup_dir`.

We are not checking if Settings and autosave have been copied correctly. We are also not checking `PYTHON_PATH`, could that still exist on some instruments?

And as per Freddie's comment, best to use something like `os.path.exists(path)` not `glob`. That way we can simplify the check a little bit.

File: installation_and_upgrade/ibex_install_utils/tasks/backup_tasks.py (hunk 1 above)
Comment (danielmaclaren) [reply installation_and_upgrade/ibex_install_utils/tasks/backup_tasks.py]: We changed it back to os.path.exists()

**Review by NikolaRoev — APPROVED** [review]

**Pull request merged**